ostd/src/arch/riscv/cpu/mod.rs

```diff
@@ -1,230 +1,19 @@
 // SPDX-License-Identifier: MPL-2.0
 
-//! CPU
+//! CPU context & state control and CPU local memory.
 
+pub mod context;
 pub mod local;
 
-use core::fmt::Debug;
-
-use riscv::register::scause::{Exception, Trap};
-
-pub use super::trap::GeneralRegs as RawGeneralRegs;
-use super::trap::{TrapFrame, UserContext as RawUserContext};
-use crate::user::{ReturnReason, UserContextApi, UserContextApiInternal};
-
-/// Cpu context, including both general-purpose registers and FPU state.
-#[derive(Clone, Copy, Debug)]
-#[repr(C)]
-pub struct UserContext {
-    user_context: RawUserContext,
-    trap: Trap,
-    fpu_state: (), // TODO
-    cpu_exception_info: CpuExceptionInfo,
-}
-
-/// CPU exception information.
-#[derive(Clone, Copy, Debug)]
-#[repr(C)]
-pub struct CpuExceptionInfo {
-    /// The type of the exception.
-    pub code: Exception,
-    /// The error code associated with the exception.
-    pub page_fault_addr: usize,
-    pub error_code: usize, // TODO
+/// Halts the CPU.
+///
+/// This function halts the CPU until the next interrupt is received. By
+/// halting, the CPU might consume less power. Internally it is implemented
+/// using the `wfi` instruction.
+///
+/// Since the function sleeps the CPU, it should not be used within an atomic
+/// mode ([`crate::task::atomic_mode`]).
+pub fn sleep_for_interrupt() {
```

Comment:
Add `#[track_caller]` to make the atomic mode break panic report to the caller of the `sleep_for_interrupt` method. Same for the `x86_64` case.